james-publisher-aws-kinesis/src/main/java/com/tomtom/james/publisher/kinesis/JSONEventFormatter.java

```diff
@@ -41,12 +43,15 @@
 
     String format(Event evt) {
         LinkedHashMap<String, Object> result = new LinkedHashMap<>();
+        long scriptDurationNano = Duration.between(evt.getTimestamp(), Instant.now()).toNanos();
```

Comment:
it is not script execution, it is script + whole publication path as you get it in kinesis plugin

Reply:
also this change does not have any test, which would be nice to see